pyomo/contrib/parmest/tests/test_utils.py

```diff
@@ -60,6 +83,125 @@ def test_convert_param_to_var(self):
+        orig_var_vals = np.array([pyo.value(v) for v in var_list])
+        orig_suffix_val = np.array([tag for _, tag in suffix_obj.items()])
+        new_vals = orig_var_vals + 0.5
+        # Update the model from the suffix
+        update_model_from_suffix(suffix_obj, new_vals)
+        # ── Check results ────────────────────────────────────────────────────
+        new_var_vals = np.array([pyo.value(v) for v in var_list])
+        new_suffix_val = np.array(list(suffix_obj.values()))
+        # (1) Variables have been overwritten with `new_vals`
+        self.assertTrue(np.allclose(new_var_vals, new_vals))
+        # (2) Suffix tags are unchanged
+        self.assertTrue(np.array_equal(new_suffix_val, orig_suffix_val))
+
+    def test_update_model_from_suffix_measurement_error(self):
+        experiment = FullReactorExperiment(data_ex, 10, 3)
+        test_model = experiment.get_labeled_model()
+
+        suffix_obj = test_model.measurement_error  # a Suffix
+        var_list = list(suffix_obj.keys())  # components
+        orig_var_vals = np.array([suffix_obj[v] for v in var_list])
+        new_vals = orig_var_vals + 0.5
+        # Update the model from the suffix
+        update_model_from_suffix(suffix_obj, new_vals)
+        # ── Check results ────────────────────────────────────────────────────
+        new_var_vals = np.array([suffix_obj[v] for v in var_list])
+        # (1) Variables have been overwritten with `new_vals`
+        self.assertTrue(np.allclose(new_var_vals, new_vals))
+
+    def test_update_model_from_suffix_length_mismatch(self):
+        m = pyo.ConcreteModel()
+
+        # Create a suffix with a Var component
+        m.x = pyo.Var(initialize=0.0)
+        m.unknown_parameters = pyo.Suffix(direction=pyo.Suffix.LOCAL)
+        m.unknown_parameters[m.x] = 0.0  # tag a Var
+        with self.assertRaisesRegex(
+            ValueError, "values length does not match suffix length"
+        ):
+            # Attempt to update with a list of different length
+            update_model_from_suffix(m.unknown_parameters, [42, 43, 44])
+
+    def test_update_model_from_suffix_not_numeric(self):
+        m = pyo.ConcreteModel()
+
+        # Create a suffix with a Var component
+        m.x = pyo.Var(initialize=0.0)
+        m.y = pyo.Var(initialize=1.0)
+        bad_value = "not_a_number"
+        m.unknown_parameters = pyo.Suffix(direction=pyo.Suffix.LOCAL)
+        # Make multiple values
+        m.unknown_parameters[m.x] = 0.0  # tag a Var
+        m.unknown_parameters[m.y] = bad_value  # tag a Var with a bad value
+        # Attempt to update with a list of mixed types
+        # This should raise an error because the suffix expects VarData or ParamData
+
+        with self.assertRaisesRegex(
+            ValueError, f"could not convert string to float: '{bad_value}'"
+        ):
+            # Attempt to update with a list of different length
+            update_model_from_suffix(m.unknown_parameters, [42, bad_value])
```

Comment:
```suggestion
        # This should raise an error because this utility only allows numeric values
        # in the model to be updated.

        with self.assertRaisesRegex(
            ValueError, f"could not convert string to float: '{bad_value}'"
        ):
            # Attempt to update with a non-numeric value
            update_model_from_suffix(m.unknown_parameters, [42, bad_value])
```